Client/SafeBrowsing/GSafeBrowsingDatabase.swift

```diff
@@ -0,0 +1,509 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+import CommonCrypto
+import Shared
+
+private let log = Logger.browserLogger
+
+extension Data {
+    func rawString() -> String {
+        if count > 0 {
+            let hexChars = Array("0123456789ABCDEF".utf8) as [UInt8]
+            return withUnsafeBytes { bytes -> String in
+                var output = [UInt8](repeating: 0, count: bytes.count * 2 + 1)
+                var index: Int = 0
+                for byte in bytes {
+                    let hi  = Int((byte & 0xF0) >> 0x04)
+                    let low = Int(byte & 0x0F)
+                    output[index] = hexChars[hi]
+                    output[index + 1] = hexChars[low]
+                    index += 0x02
+                }
+                return String(cString: &output)
+            }
+        }
+        return ""
+    }
+}
+
+extension String {
+    func rawData() -> Data {
+        let length = self.count
+        if length & 1 != 0 {
+            return Data()
+        }
+        
+        var data = Data()
+        data.reserveCapacity(length / 2)
+        var index = self.startIndex
+        for _ in 0..<length / 2 {
+            let nextIndex = self.index(index, offsetBy: 2)
+            if let byte = UInt8(self[index..<nextIndex], radix: 16) {
+                data.append(byte)
+            } else {
+                return Data()
+            }
+            index = nextIndex
+        }
+        return data
+    }
+}
+
+class SafeBrowsingDatabase {
+    private let dbLock = NSRecursiveLock()
```

Comment:
I don't think we need that for core data with proper setup

Reply:
It is needed for IF the database gets corrupted. When that happens (which can happen on a background thread), I need to clear the entire DB and re-create it.

The fastest way I found to do this is to just delete the DB instead of going through and removing all objects. It also gives the DB a cleaner/fresh state.